### PR TITLE
VP/BE-118. Transfer tokens between users.

### PR DESCRIPTION
**Pull request by dev-vadim**
**Description:**

Implement transfer wrapped tokens between users;
Add Operation::Transfer;
Add transfer tests;

### CODE REVIEW

File: pallets/minterest-protocol/src/lib.rs [hunk 1]
```diff
@@ -528,4 +550,37 @@ impl<T: Trait> Module<T> {
...
+	/// - `transfer_amount`: the amount of the wrapped asset to transfer.
+	fn do_transfer(
+		who: &T::AccountId,
+		receiver: &T::AccountId,
+		wrapped_id: CurrencyId,
+		transfer_amount: Balance,
+	) -> DispatchResult {
+		ensure!(transfer_amount > Balance::zero(), Error::<T>::ZeroBalanceTransaction);
+		ensure!(who != receiver, Error::<T>::CannotTransferToSelf);
+
```
Comment (lut1k) [inline pallets/minterest-protocol/src/lib.rs:568]: Add pause operation check:
1. Primitives -> enum Operation add Transfer;
2. Fix file types.json;
3. Fix pallet Controller (Events, extrinsic, unit-tests and other ...);

File: pallets/minterest-protocol/src/tests.rs [hunk 2]
```diff
@@ -616,3 +616,103 @@ fn disable_collateral_should_work() {
...
+				Error::<Test>::NotValidWrappedTokenId
+			);
+
+			// Alice is unable to transfer tokens to self
+			assert_noop!(
+				TestProtocol::transfer_wrapped(alice(), ALICE, CurrencyId::MDOT, ONE_HUNDRED_DOLLARS),
+				Error::<Test>::CannotTransferToSelf
+			);
+
+			// Alice is unable to transfer more tokens tan she has
```
Comment (lut1k) [inline pallets/minterest-protocol/src/tests.rs:700]: 'than' she has

File: pallets/minterest-protocol/src/tests.rs [hunk 3]
```diff
@@ -616,3 +616,103 @@ fn disable_collateral_should_work() {
...
+			assert_noop!(
+				TestProtocol::transfer_wrapped(alice(), BOB, CurrencyId::MDOT, dollars(101_u128)),
+				Error::<Test>::NotEnoughWrappedTokens
+			);
+
+			// Alice is unable to transfer tokens with zero balance
+			assert_noop!(
+				TestProtocol::transfer_wrapped(bob(), ALICE, CurrencyId::MDOT, 1_u128),
+				Error::<Test>::NotEnoughWrappedTokens
+			);
```
Comment (lut1k) [inline pallets/minterest-protocol/src/tests.rs:710]: the comment does not match the test

File: pallets/minterest-protocol/src/tests.rs [hunk 4]
```diff
@@ -616,3 +616,103 @@ fn disable_collateral_should_work() {
...
+
+			// Bob is unable to send zero tokens
+			assert_noop!(
+				TestProtocol::transfer_wrapped(bob(), ALICE, CurrencyId::MBTC, Balance::zero()),
+				Error::<Test>::ZeroBalanceTransaction
+			);
+		});
+}
```
Comment (lut1k) [inline pallets/minterest-protocol/src/tests.rs:718]: You need to add a test for Error::<T>::RedeemControllerRejection.
To do this, you can implement an integration test on a pallet integration-tests -> minterest_protocol_tests.rs.

**Review by lut1k — DISMISSED** [review]

File: pallets/integration-tests/src/tests/minterest_protocol_tests.rs [hunk 5]
```diff
@@ -1254,4 +1254,594 @@ mod tests {
...
+	// It is possible to transfer assets from the pool insurance.
+	// 1. Deposit 10 DOT to pool insurance;
+	// 2. Alice deposit 20 DOT;
+	// 3. Bob deposit 20 BTC;
+	// 4. Bob deposit 10 DOT;
+	// 5. Bob borrow 15 DOT;
+	// 6. Alice `transfer_wrapped` 20 MDOT;
+	// 7. DOT pool insurance equal 5 DOT;
+	#[test]
+	fn transfer_wrapped_over_insurance() {
```
Comment (Apollo50) [inline pallets/integration-tests/src/tests/minterest_protocol_tests.rs:1745]: In my view, Test can be removed cause this case is already checked in previously test's algorithm.

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**